### PR TITLE
fix: Exclude deleted price lists and prices from pricing calculations

### PR DESCRIPTION
**Pull request by panalgin**
Hello,

## Description of the bug
A bug was discovered in the pricing calculation logic where deleted price lists and prices were still being considered, potentially leading to incorrect pricing information.

## Fix implementation
The fix adds two conditions:
1. A condition to the left join in the `calculatePrices` method to ensure only non-deleted price lists are considered.
2. A where clause to the base price table select function to exclude deleted prices.

### Code changes
1. Price List Join:
```typescript
.leftJoin("price_list as pl", function () {
  this.on("pl.id", "price.price_list_id").andOn(
    "pl.status",
    knex.raw("?", [PriceListStatus.ACTIVE])
  ).andOn(knex.raw("pl.deleted_at IS NULL"))
})
```

2. Base Price Table Select:
```typescript
.where(knex.raw("price.deleted_at IS NULL"))
```

## Impact and importance
These fixes are crucial for accurate pricing calculations, ensuring that only active, non-deleted price lists and prices are considered.

## Testing
Unable to write test cases due to module tests not working, but the fixes have been manually verified.

## Additional notes
- No new dependencies or API changes.
- Minimal changes addressing specific edge cases.
- Future testing improvements are needed.
- These changes prevent potential issues with pricing calculations that could arise from considering deleted data.

I'm open to any feedback or suggestions for improvement. Thank you for considering this pull request.

Deleting a price list from this screen:
image

In the database, the record is deleted but because its status still active, and no-check for deleted_at
image

### CODE REVIEW

**Review by carlos-r-l-rodrigues — APPROVED** [review]
LGTM

**Review by riqwan — APPROVED** [review]
good catch! @panalgin

File: packages/modules/pricing/src/repositories/pricing.ts [hunk 1]
```diff
@@ -81,7 +81,7 @@ export class PricingRepository
         this.on("pl.id", "price.price_list_id").andOn(
           "pl.status",
           knex.raw("?", [PriceListStatus.ACTIVE])
-        )
+        ).andOn(knex.raw("pl.deleted_at IS NULL"))
```
Comment (riqwan) [inline packages/modules/pricing/src/repositories/pricing.ts:85]: If you can add a similar where clause on the price table, that will be helpful!

Comment (panalgin) [reply packages/modules/pricing/src/repositories/pricing.ts:85]: Sure I will do!

Comment (panalgin) [reply packages/modules/pricing/src/repositories/pricing.ts:85]: [704f8c4](https://github.com/medusajs/medusa/pull/9015/commits/704f8c459085580c32a7fbdc46951066975d5d50)

I think this filters it, tested on production

Comment (riqwan) [reply packages/modules/pricing/src/repositories/pricing.ts:85]: yup, that looks good! Thanks for the fix!

**Review by olivermrbl — APPROVED** [review]
legend @panalgin – thanks for the contribution

**Comment by Attachai77** [comment]
Could you merge this ?

**Comment by fPolic** [comment]
This was fixed in the meantime as part of https://github.com/medusajs/medusa/pull/9265.

**Comment by olivermrbl** [comment]
@fPolic, can we close this?

**Comment by fPolic** [comment]
Yes
